tests/integration.rs

```diff
@@ -5490,6 +5490,39 @@ mod tests {
                     fname
                 ))
             }
+
+            fn wait_for_boot(&self) -> bool {
+                let cmd = "dir /b c:\\ | find \"Windows\"";
+                let tmo_max = 180;
+                // The timeout increase by n+n*(i+1)+..., therefore the initial
+                // interval must be small.
+                let tmo_int = 2;
+                let out = std::panic::catch_unwind(|| {
+                    match ssh_command_ip_with_auth(
+                        cmd,
+                        &self.auth,
+                        &self.guest.network.guest_ip,
+                        {
+                            let mut ret = 1;
+                            while tmo_int * ret < tmo_max {
+                                ret += 1;
+                            }
+                            ret
+                        },
```

Comment:
I don't quite follow why we need the loop here. I think this loop is simply calculating `tmo_max/tmo_int`, assuming there is no remainder (for this case `180/2`).

Reply:
The loop calculates the number of iterations necessary to keep total timeout to be around`tmo_max`, please see  [ssh_command_ip_with_auth](https://github.com/cloud-hypervisor/cloud-hypervisor/blob/master/test_infra/src/lib.rs#L571) on how the sleep time grows `timeout*counter; counter++`. As we've arranged not to change the implementation of `ssh_command_ip_with_auth`, the number of retry needs to be reverse calculated considering that. Like in this case, with the starting timeout value of 2 there would be around 14 iterations to reach 180.

Thanks

Reply:
```python
In [39]: tmo_int = 2; tmo_max = 180; i = 1; tmo_i = 0

In [40]: while tmo_i < tmo_max:
    ...:     tmo_i += tmo_int * i; i += 1; print(tmo_i)
    ...:
2
6
12
20
30
42
56
72
90
110
132
156
182
```

Reply:
Yes, I think we are on the same page for the behavior of `ssh_command_ip_with_auth()`. And I know understand what you intended to do, having a total timeout as `180s`. 

But I think the code in the PR is not doing what you want:

```
// let tmo_max = 180;
// let tmo_int = 2;
{
    let mut ret = 1;
    while tmo_int * ret < tmo_max {  // "tmo_int" is never changed (major difference from your example above)
       ret += 1;
    }
    ret
 } // This block will always be evaluated to "90"
```

Reply:
I missed the forest for the trees :) Fixed, thanks a lot!